runtime/commonMain/src/kotlinx/serialization/internal/Enums.kt

```diff
@@ -5,13 +5,17 @@
 package kotlinx.serialization.internal
 
 import kotlinx.serialization.*
+import kotlin.jvm.JvmOverloads
 import kotlin.reflect.KClass
 
-public class EnumDescriptor(override val name: String, private val choices: Array<String>) : SerialClassDescImpl(name) {
+internal class EnumDescriptor @JvmOverloads constructor(
```

Comment:
Do you still need JvmOverloads here?

Reply:
I just double-checked and found that `EnumDescriptor` is referenced from the generated code. So we should go with `@InternalSerializationApi public EnumDescriptor`. Seems it was working by accident that `internal` classes are available from test sources of the same module.

Reply:
>internal classes are available from test sources of the same module.

Nope, that's actually what friend modules are for right now